cmd/soroban-rpc/lib/preflight/src/fees.rs

```diff
@@ -304,51 +307,42 @@ pub(crate) fn compute_bump_footprint_exp_transaction_data_and_min_fee(
     ledger_storage: &ledger_storage::LedgerStorage,
     bucket_list_size: u64,
     current_ledger_seq: u32,
-) -> Result<(SorobanTransactionData, i64), Box<dyn error::Error>> {
-    let mut rent_changes: Vec<LedgerEntryRentChange> = Vec::new();
-    for key in (&footprint).read_only.as_vec() {
-        let unmodified_entry = ledger_storage.get(key, false)?;
-        let size = (key.to_xdr()?.len() + unmodified_entry.to_xdr()?.len()) as u32;
-        let expirable_entry: Box<dyn ExpirableLedgerEntry> = (&unmodified_entry).try_into()?;
-        let new_expiration_ledger = current_ledger_seq + ledgers_to_expire;
-        if new_expiration_ledger <= expirable_entry.expiration_ledger_seq() {
-            // The bump would be ineffective
-            continue;
-        }
-        let rent_change = LedgerEntryRentChange {
-            is_persistent: expirable_entry.durability() == Persistent,
-            old_size_bytes: size,
-            new_size_bytes: size,
-            old_expiration_ledger: expirable_entry.expiration_ledger_seq(),
-            new_expiration_ledger,
-        };
-        rent_changes.push(rent_change);
-    }
+) -> Result<(SorobanTransactionData, i64)> {
+    let rent_changes = compute_bump_footprint_rent_changes(
+        &footprint,
+        ledger_storage,
+        ledgers_to_expire,
+        current_ledger_seq,
+    )
+    .context("cannot compute rent changes")?;
```

Comment:
Could you modify the error message to uniquely identify the error ? i.e. so that it wouldn't be the same error text for both bump and restore ? i.e. `cannot compute bump rent changes`

Reply:
Sure, although the operation passed will be different